archaius2-guice/src/main/java/com/netflix/archaius/guice/ArchaiusModule.java

```diff
@@ -15,299 +15,251 @@
+        this.remoteLayer      = new DefaultCompositeConfig();
+        try {
+            this.rawConfig       = DefaultCompositeConfig.builder()
+                    .withConfig(RUNTIME_LAYER_NAME,      runtimeLayer)
+                    .withConfig(REMOTE_LAYER_NAME,       remoteLayer)
+                    .withConfig(SYSTEM_LAYER_NAME,       SystemConfig.INSTANCE)
+                    .withConfig(ENVIRONMENT_LAYER_NAME,  EnvironmentConfig.INSTANCE)
+                    .withConfig(APPLICATION_LAYER_NAME,  applicationLayer)
+                    .withConfig(LIBRARIES_LAYER_NAME,    librariesLayer)
+                    .build();
+        } catch (ConfigException e) {
+            throw new ProvisionException("Error creating raw configuration", e);
+        }
+    }
+    
+    public ArchaiusModule withConfigName(String value) {
+        this.configName = value;
+        return this;
+    }
+    
+    public ArchaiusModule withApplicationOverrides(Properties prop) {
+        return withApplicationOverrides(MapConfig.from(prop));
+    }
     
-    private static final String RUNTIME_LAYER_NAME              = "RUNTIME";
-    private static final String REMOTE_LAYER_NAME               = "REMOTE";
-    private static final String SYSTEM_LAYER_NAME               = "SYSTEM";
-    private static final String ENVIRONMENT_LAYER_NAME          = "ENVIRONMENT";
-    private static final String APPLICATION_OVERRIDE_LAYER_NAME = "APPLICATION_OVERRIDE";
-    private static final String APPLICATION_LAYER_NAME          = "APPLICATION";
-    private static final String LIBRARIES_LAYER_NAME            = "LIBRARIES";
-    private static final String DEFAULTS_LAYER_NAME             = "DEFAULTS";
+    public ArchaiusModule withApplicationOverrides(Config config) {
+        this.applicationOverrides = config;
+        return this;
+    }
     
-    private static final AtomicInteger idCounter = new AtomicInteger();
+    public ArchaiusModule withCascadeStrategy(Class<? extends CascadeStrategy> cascadeStrategy) {
+        this.cascadeStrategy = cascadeStrategy;
+        return this;
+    }
     
     @Override
     protected void configure() {
         bindListener(Matchers.any(), new ConfigurationInjectingListener());
-
+        
+        bind(CascadeStrategy.class).to(cascadeStrategy);
+        
         Multibinder.newSetBinder(binder(), ConfigReader.class)
             .addBinding().to(PropertiesConfigReader.class).in(Scopes.SINGLETON);
-        
-        bind(ArchaiusConfiguration.class).to(OptionalArchaiusConfiguration.class);
     }
-    
+
     @Provides
     @Singleton
     @RuntimeLayer
     SettableConfig getSettableConfig() {
-        return new DefaultSettableConfig();
-    }
-    
-    @Provides
-    @Singleton
-    @DefaultsLayer
-    SettableConfig getDefaultsConfig() {
-        return new DefaultSettableConfig();
+        return runtimeLayer;
     }
     
     @Provides
     @Singleton
-    @ApplicationLayer 
-    CompositeConfig getApplicationLayer() {
```

Comment:
I'm trying to reduce the API so it only exposes things are should be consumed.  I don't think the @ApplicationLayer is something that should be consumed.